### PR TITLE
@W-11261325@: v3 heartbeat job workflow [2.x]

### PR DESCRIPTION
**Pull request by gkjung**
Heartbeat job that runs on release-3 latest-pilot and does not trigger any critical alerts.

### CODE REVIEW

File: .github/workflows/production-heartbeat-v3-test.yml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+          if [[ -n "${{ secrets.CLI_VERSION }}" ]]; then
+            echo "CLI_VERSION=@${{ secrets.CLI_VERSION}}" >> $GITHUB_ENV
+          fi
+          # If the SCANNER_VERSION repo secret is set, we want to install that version of sfdx-scanner, so we set an
+          # environment variable. Otherwise, we use `latest-pilot` to run the latest version of v3.
+          # Note: This can be used to intentionally fail the GHA by providing an invalid version number.
+          if [[ -n "${{ secrets.SCANNER_VERSION }}" ]]; then
+            echo "SCANNER_VERSION=@${{ secrets.SCANNER_VERSION }}" >> $GITHUB_ENV
+          else
+            echo "SCANNER_VERSION=@latest-pilot" >> $GITHUB_ENV
```
Comment (gkjung) [inline .github/workflows/production-heartbeat-v3-test.yml:49]: added this to make the default v3 instead

File: .github/workflows/production-heartbeat-v3.yml [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+          if [[ -n "${{ secrets.CLI_VERSION }}" ]]; then
+            echo "CLI_VERSION=@${{ secrets.CLI_VERSION}}" >> $GITHUB_ENV
+          fi
+          # If the SCANNER_VERSION repo secret is set, we want to install that version of sfdx-scanner, so we set an
+          # environment variable. Otherwise, we use `latest-pilot` to run the latest version of v3.
+          # Note: This can be used to intentionally fail the GHA by providing an invalid version number.
+          if [[ -n "${{ secrets.SCANNER_VERSION }}" ]]; then
+            echo "SCANNER_VERSION=@${{ secrets.SCANNER_VERSION }}" >> $GITHUB_ENV
+          else
+            echo "SCANNER_VERSION=@latest-pilot" >> $GITHUB_ENV
```
Comment (gkjung) [inline .github/workflows/production-heartbeat-v3.yml:51]: added this to make the default v3 instead

File: .github/workflows/production-heartbeat-v3.yml [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+      matrix:
+        os: [{vm: ubuntu-latest, exe: .sh}, {vm: windows-2019, exe: .cmd}]
+        node: ['lts/*']
+    runs-on: ${{ matrix.os.vm }}
+    timeout-minutes: 60
+    steps:
+      # === Setup. We need to get the code, set up nodejs, and create the results directory. ===
+      - uses: actions/checkout@v2
+        with:
+          ref: 'release-3'
```
Comment (gkjung) [inline .github/workflows/production-heartbeat-v3.yml:24]: Changed from release to release-3

File: .github/workflows/production-heartbeat-v3.yml [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+          # Build the status strings for each step as environment variables to save space later. Null retry_count values
+          # will be replaced with `n/a` to maintain readability in the alert.
+          CLI_INSTALL_STATUS: ${{ steps.sfdx_install.outcome }} after ${{ steps.sfdx_install.outputs.retry_count || 'n/a' }} retries
+          SCANNER_INSTALL_STATUS: ${{ steps.scanner_install.outcome }} after ${{ steps.scanner_install.outputs.retry_count || 'n/a' }} retries
+          SMOKE_TESTS_STATUS: ${{ steps.smoke_tests.outcome }}
+          # A link to this run, so the PagerDuty assignee can quickly get here.
+          RUN_LINK: https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}
+        run: |
+          # GHA env-vars don't have robust conditional logic, so we'll use this if-else branch to define some bash env-vars.
+          ALERT_SEV="info" # a failure should not throw a critical alert
```
Comment (gkjung) [inline .github/workflows/production-heartbeat-v3.yml:122]: Severity will always be "info" even if it fails

**Review by jfeingold35 — APPROVED** [review]

**Pull request merged**